sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -538,3 +538,14 @@ case class UnresolvedOrdinal(ordinal: Int)
   override def nullable: Boolean = throw new UnresolvedException(this, "nullable")
   override lazy val resolved = false
 }
+
+/**
+ * Represents unresolved aggregate with having clause, it is turned by the analyzer into a Filter.
+ */
+case class AggregateWithHaving(
```

Comment:
`UnresolvedHaving`?

Reply:
We also discussed the naming here: https://github.com/apache/spark/pull/28294#discussion_r413424305

Reply:
Does `GroupingSets` have this bug as well?

Reply:
I think `GroupingSets` don't have the same bug here.
But I test the 2 below queries:
```
select sum(a) as b, '2020-01-01' as fake
FROM VALUES (1, 10), (2, 20) AS T(a, b)
group by GROUPING SETS ((b), (a, b))
having b > 10;
```
and adding the cast
```
select sum(a) as b, cast('2020-01-01' as date) as fake
FROM VALUES (1, 10), (2, 20) AS T(a, b)
group by GROUPING SETS ((b), (a, b))
having b > 10;
```
Both queries return empty results, it seems the HAVING after GROUPING SET resolve expression in select clauses first. Maybe it's another bug, I think we should follow the resolving strategy of table scheme first and then select clause?